sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
Why is this condition here? Why use absolute 9 rather than the define below?

Does a 0 value turn off keep alives?